src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java

```diff
@@ -0,0 +1,188 @@
+    }
+    
+    @Test
+    public void testAchFileHeaderDateIsOneDayAfterProcessDate() {
+        cuExtractPaymentServiceImpl.setAchBundlerHelperService(new TestAchBundlerHelperService());
+        
+        PaymentStatus extractedStatus = new PaymentStatus();
+        Date processDate = new Date(Calendar.getInstance().getTimeInMillis());
+        List<String> notificationEmailAddresses = new ArrayList<String>(Arrays.asList(NOTIFICATION_EMAIL_ADDRESSES));
+        
+        cuExtractPaymentServiceImpl.writeExtractAchFileMellonBankFastTrack(extractedStatus, ACH_EXTRACT_TEST_PATH_AND_FILE, processDate, yyyyMMddHHmmss_DATE_FORMATTER, notificationEmailAddresses);
+        
+        Date expectedHeaderDate = cuExtractPaymentServiceImpl.calculateHeaderDate(processDate);
+        assertTrue("ACH Extract Payments File Header Date is not what was expected.", isDateInOutputFileWhereExpected(HEADER_LINE_IN_OUTPUT_FILE, yyyyMMddHHmmss_DATE_FORMATTER.format(expectedHeaderDate)));
+        assertTrue("ACH Extract Payments File Payee Date is not what was expected.", isDateInOutputFileWhereExpected(FIRST_PAYEE_LINE_IN_OUTPUT_FILE, yyyyMMdd_DATE_FORMATTER.format(processDate)));
+    }
+    
+    private boolean isDateInOutputFileWhereExpected(int numberOfLinesToRead, String formattedDateExpected) {
+        BufferedReader inputReader = null;
+        boolean expectedWasFound = false;
+        try {
+            String lineWeWant = null;
+            inputReader = new BufferedReader(new FileReader(new String (ACH_EXTRACT_TEST_PATH_AND_FILE_FOR_VALIDATION)));
+            for (int i=0; i<numberOfLinesToRead; i++) {
+                lineWeWant = inputReader.readLine();
+            }
+            expectedWasFound = lineWeWant.contains(formattedDateExpected);
+        }
+        catch (IOException ie) {
+            expectedWasFound = false;
+        }
+        finally {
+            if (inputReader != null) {
+                try {
+                    inputReader.close();
+                }
+                catch (IOException ie) {
+                    expectedWasFound = false;
+                }
+            }
+        }
+        return expectedWasFound;
+    }
+
+    private class TestCuExtractPaymentServiceImpl extends CuExtractPaymentServiceImpl {
+        @Override
+        protected boolean isProduction() {
+            return false;
+        }
+    }
+
+    private class TestAchBundlerHelperService implements AchBundlerHelperService {
+
+        private static final String BANKCODE_1 = "bankCode1";
+        private final Integer DISB_ID_1 = new Integer(1);
+
+        @Override
+        public Iterator<PaymentDetail> getPendingAchPaymentDetailsByDisbursementNumberAndBank(Integer disbursementNumber, String bankCode) {
+            ArrayList<PaymentDetail> paymentDetails = new ArrayList<PaymentDetail>();
+
+            PaymentDetail paymentDetail_1 = new PaymentDetail();
```

Comment:
I would leave off the suffixes and refactor these names if/when future expansion of this unit test necessitates it. In general, I prefer to avoid coding for future expansion when first introducing code because often the future expansion never occurs or when it does the needs are different and the original code ends up adding unintentional technical debt.